components/tools/OmeroJava/test/integration/AbstractServerTest.java

```diff
@@ -285,6 +298,174 @@ public void tearDown() throws Exception {
+    }
+
+    /**
+     * Create a link between a Dataset and an Image.
+     * @param dataset an OMERO Dataset
+     * @param image an OMERO Image
+     * @return the created link
+     * @throws ServerError an error possibly occurring during saving of the link
+     */
+    protected DatasetImageLink linkParentToChild(Dataset dataset, Image image) throws ServerError {
+        if (dataset.isLoaded() && dataset.getId() != null) {
+            dataset = (Dataset) dataset.proxy();
+        }
+        if (image.isLoaded() && image.getId() != null) {
+            image = (Image) image.proxy();
+        }
+
+        final DatasetImageLink link = new DatasetImageLinkI();
+        link.setParent(dataset);
+        link.setChild(image);
+        return (DatasetImageLink) iUpdate.saveAndReturnObject(link);
+    }
+
+    /**
+     * Assert that the given object is in the given group.
+     * @param object a model object
+     * @param group an ExperimenterGroup
+     * @throws ServerError unexpected
+     */
+    protected void assertInGroup(IObject object, ExperimenterGroup group) throws ServerError {
+        assertInGroup(Collections.singleton(object), group.getId().getValue());
+    }
+
+    /**
+     * Assert that the given object is in the given group.
+     * @param object a model object
+     * @param expectedGroupId a group Id
+     * @throws ServerError unexpected
+     */
+    protected void assertInGroup(IObject object, long expectedGroupId) throws ServerError {
+        assertInGroup(Collections.singleton(object), expectedGroupId);
+    }
+
+    /**
+     * Assert that the given objects are in the given group.
+     * @param objects some model objects
+     * @param expectedGroupId a group Id
+     * @throws ServerError unexpected
+     */
+    protected void assertInGroup(Collection<? extends IObject> objects, long expectedGroupId) throws ServerError {
+        verifyObjectProperty(objects, expectedGroupId, Properties.GROUP);
+    }
+
+    /**
+     * Assert that certain objects either belong to a certain group
+     * or have a certain owner.
+     * @param testedObjects some model objects to test for properties
+     * @param id expected id of the property object (of GROUP or OWNER)
+     * @param property property to examine the testedObjects for (can be GROUP or OWNER)
+     * @throws ServerError if query fails
+     */
+    protected void verifyObjectProperty(Collection<? extends IObject> testedObjects, long id, Properties property) throws ServerError {
+        if (testedObjects.isEmpty()) {
+            throw new IllegalArgumentException("must assert about some objects");
+        }
+        for (final IObject testedObject : testedObjects) {
+            final String testedObjectName = testedObject.getClass().getName() + '[' + testedObject.getId().getValue() + ']';
+            final String query = "SELECT details." + property.toString() + ".id FROM " + testedObject.getClass().getSuperclass().getSimpleName() +
```

Comment:
You do need it. :smiley: In Java, `someString + someObject` always implicitly becomes `someString + someObject.toString()`.